x-pack/plugins/observability_ai_assistant/public/hooks/use_timeline.ts

```diff
@@ -257,26 +258,27 @@ export function useTimeline({
     });
   }
 
-  const items = useMemo(() => {
-    if (pendingMessage) {
+  const itemsWithAddedLoadingStates = useMemo(() => {
+    // While we're loading we add an empty loading chat item:
```

Comment:
I'm not sure if we want to add an empty chat item. If it's a function call, the item will collapse again. I think previously we had an approach of: the last timeline item is marked as loading if we are waiting on _something_, e.g. a function call being executed.

Reply:
Yes, but the last item that is loading can be in a collapsed group. That caused the loading message suddenly disappearing while a function call was in flight.

Reply:
But you can still add a loading state to the element that displays the collapsed items, no? Or am I misunderstanding what you mean?

Reply:
The item that renders the Collapsed Items doesn't add a `ChatTimelineItem` to the list that has `loading: true`. `useTimeline` is currently responsible for adding all additional loading `ChatTimelineItem`s to the array. This was already the case before my changes. 

Are you saying you want `<ChatCollapsedItems />` to add  an additional loading `ChatTimelineItem` while one of the items inside the collapsed items are loading?

Reply:
I'm not particularly concerned with _how_ we do it, but essentially the avatar of the last item in the timeline should display a loading spinner - does that make sense?

Reply:
Yes, that works right now.

Reply:
I think it's slightly different, in that there's always a chat item. I think that's okay though, there should always be a visual or a text response from the LLM eventually.